userspace/falco/app_actions/configure_interesting_sets.cpp

```diff
@@ -15,27 +15,106 @@ limitations under the License.
 */
 
 #include "application.h"
+#include "configure_interesting_sets.h"
+#include <unordered_set>
+#include <sinsp.h>
+#include <sstream>
 
 using namespace falco::app;
 
-void application::configure_interesting_sets()
+extern sinsp_evttables g_infotables;
+
+std::string concat_evttypes_names(std::unordered_set<std::string> const names)
 {
-	/// TODO: in the next future we need to change the interface of `enforce_simple_ppm_sc_set`
-	/// and `enforce_sinsp_state_tp` APIs, they shouldn't require an inspector to be called!
-	std::unique_ptr<sinsp> inspector(new sinsp());
+	std::stringstream ss;
+	std::copy(names.begin(), names.end(),
+	std::ostream_iterator<std::string>(ss, ", "));
+	std::string names_str = ss.str();
+	return names_str.substr(0, names_str.size() - 2);
+}
 
-	/* Please note: here we fill these 2 sets because we are interested in only some features, if we leave
-	 * them empty `libsinsp` will fill them with all the available syscalls and all the available tracepoints!
-	 */
+void application::extract_rules_event_names(std::unique_ptr<sinsp>& inspector, std::unordered_set<std::string>& rules_evttypes_names)
+{
+	/* Get all PPME events from all rules as idx codes. */
+	std::set<uint16_t> rule_events;
+	std::string source = falco_common::syscall_source;
+	m_state->engine->evttypes_for_ruleset(source, rule_events);
+	std::unordered_set<uint32_t> ppme_events_codes(rule_events.begin(), rule_events.end());
+
+	/* Translate PPME event codes to event names. */
+	rules_evttypes_names = inspector->get_events_names(ppme_events_codes);
```

Comment:
We might want to add a new sinsp API to retrieve ppm_sc set from given events_set, instead of using event names ;) 
Perhaps there is already such an api; i can't recall.
@Andreagit97

Reply:
In general, would love having proper `sinsp` APIs for everything, so that here we don't have to loop anywhere and just call the APIs. Depending on timing we could merge and then cleanup iteratively afterwards in follow up PRs.